domain/network/state/subnet.go

```diff
@@ -381,6 +381,29 @@ WHERE  subnet_cidr = $M.cidr`
 	return resultSubnets.ToSubnetInfos(), nil
 }
 
+// getSubnetByProviderID retrieves subnet information for a
+// specific provider ID using the provided transaction and context.
+func (st *State) getSubnetByProviderID(ctx context.Context, tx *sqlair.TX, id string) (*network.SubnetInfo, error) {
+
+	row := SubnetRow{ProviderID: id}
+	stmt, err := st.Prepare(`
+SELECT &SubnetRow.*
+FROM   v_space_subnet
+WHERE  subnet_provider_id = $SubnetRow.subnet_provider_id`, row)
+	if err != nil {
+		return nil, errors.Errorf("preparing get subnet by provider id statement: %w", err)
+	}
+
+	if err := tx.Query(ctx, stmt, row).Get(&row); err != nil {
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil, networkerrors.SubnetNotFound
+		}
+		return nil, errors.Errorf("retrieving subnet by provider id %q: %w", id, err)
```

Comment:
Have a look at what will happen as this flows up the stack. You stutter by repeating the provider ID.